client/state/happychat/chat/reducer.js

```diff
@@ -0,0 +1,181 @@
+/** @format */
+
+/**
+ * External dependencies
+ *
+ * @format
+ */
+import { concat, filter, find, map, get, sortBy, takeRight } from 'lodash';
+import validator from 'is-my-json-valid';
+
+/**
+ * Internal dependencies
+ */
+import {
+	SERIALIZE,
+	DESERIALIZE,
+	HAPPYCHAT_SEND_MESSAGE,
+	HAPPYCHAT_SET_MESSAGE,
+	HAPPYCHAT_RECEIVE_EVENT,
+	HAPPYCHAT_SET_CHAT_STATUS,
+	HAPPYCHAT_TRANSCRIPT_RECEIVE,
+} from 'state/action-types';
+import {
+	HAPPYCHAT_CHAT_STATUS_DEFAULT,
+	HAPPYCHAT_MAX_STORED_MESSAGES,
+} from 'state/happychat/constants';
+import { combineReducers } from 'state/utils';
+import { timelineSchema } from './schema';
+
+export const lastActivityTimestamp = ( state = null, action ) => {
+	switch ( action.type ) {
+		case HAPPYCHAT_SEND_MESSAGE:
+		case HAPPYCHAT_RECEIVE_EVENT:
+			return Date.now();
+	}
+	return state;
+};
+lastActivityTimestamp.schema = { type: 'number' };
+
+/**
+ * Tracks the state of the happychat chat. Valid states are:
+ *
+ *  - HAPPYCHAT_CHAT_STATUS_DEFAULT : no chat has been started
+ *  - HAPPYCHAT_CHAT_STATUS_PENDING : chat has been started but no operator assigned
+ *  - HAPPYCHAT_CHAT_STATUS_ASSIGNING : system is assigning to an operator
+ *  - HAPPYCHAT_CHAT_STATUS_ASSIGNED : operator has been connected to the chat
+ *  - HAPPYCHAT_CHAT_STATUS_MISSED : no operator could be assigned
+ *  - HAPPYCHAT_CHAT_STATUS_ABANDONED : operator was disconnected
+ *  - HAPPYCHAT_CHAT_STATUS_CLOSED : chat was closed
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ *
+ */
+export const status = ( state = HAPPYCHAT_CHAT_STATUS_DEFAULT, action ) => {
+	switch ( action.type ) {
+		case HAPPYCHAT_SET_CHAT_STATUS:
+			return action.status;
+	}
+	return state;
+};
+
+/**
+ * Tracks the current message the user has typed into the happychat client
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ *
+ */
+export const message = ( state = '', action ) => {
+	switch ( action.type ) {
+		case HAPPYCHAT_SEND_MESSAGE:
+			return '';
+		case HAPPYCHAT_SET_MESSAGE:
+			return action.message;
+	}
+	return state;
+};
+
+/**
+ * Returns a timeline event from the redux action
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ *
+ */
+const timeline_event = ( state = {}, action ) => {
```

Comment:
If we are here refactoring could we maybe change this to camelcase.

Reply:
Sure, this name is internal to this file, so it doesn't affect too much. I'll change it.